kibana-reports/public/components/context_menu/context_menu.js

```diff
@@ -234,12 +234,34 @@ $(function () {
   locationHashChanged();
 });
 
+const isDiscoverNavMenu = (navMenu) => {
+  return navMenu[0].children.length === 5 && 
+    ($('[data-test-subj="breadcrumb first"]').prop('title') === 'Discover' ||
+    $('[data-test-subj="breadcrumb first last"]').prop('title') === 'Discover')
```

Comment:
why there are 2 conditions being checked, while the other functions contain one.

Reply:
Because on the `Discover` page the Reporting menu will be present in the menu even if there is no Saved search selected, but the id of the `Discover` breadcrumb prop is different when there is a Saved search selected vs. when there isn't

Reply:
got it. So, this only happens with Discover, not the other two?

Reply:
Yeah- because on the base `Dashboard` and `Visualizations` screens it is just a list of the available ones, but you are not on the actual page with the nav menu

Reply:
Did you try the "create visualization/dashboard"? I think it leads to a similar scenario with `Discover`

Reply:
Good catch- I will have to add those checks as well